src/apps/VirtualApp/components/VerifyAccount.js

```diff
@@ -0,0 +1,196 @@
+import React, { useMemo, useState, useEffect } from 'react';
+import {
+  Button,
+  Checkbox,
+  Header,
+  Icon,
+  Input,
+  Modal,
+  Segment,
+  Table,
+} from 'semantic-ui-react';
+import { useTranslation } from 'react-i18next';
+import {
+  AUTH_API_BACKEND,
+} from "../../../shared/env";
+import {silentSignin} from "../../../components/UserManager";
+
+
+const EMAIL_RE = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+const emailValid = (email) => !!EMAIL_RE.test(String(email).toLowerCase());
+
+const VerifyAccount = (props) => {
+  const {loginPage, i18n} = props;
+  const {t} = useTranslation();
+  const {user, user: {access_token}} = props;
+  const [email, setEmail] = useState('');
+  const valid = useMemo(() => emailValid(email), [email]);
+  const [requestSent, setRequestSent] = useState(false);
+  const [closedModal, setClosedModal] = useState(false);
+  const [pendingState, setPendingState] = useState({});
+  const [error, setError] = useState('');
+  const [formClosed, setFormClosed] = useState(false);
+  const direction = i18n.language === 'he' ? 'rtl' : '';
+  const textAlign = i18n.language === 'he' ? 'right' : '';
+
+  useEffect(() => {
+    if (user.pending && user.pending.length) {
+      const updatePendingState = user.pending.reduce((acc, pending_user) => {
+        acc[pending_user] = '';
+        return acc;
+      }, {});
+      setPendingState(updatePendingState);
+    }
+  }, [user.pending]);
+
+  const updateUserPendingState = (email, state) => {
+    setPendingState(Object.assign({}, pendingState, {[email]: state}));
+  }
+
+  const applyPendingStates = () => {
+    for (let [pendingEmail, state] of Object.entries(pendingState)) {
+      if (['ignore', 'approve'].includes(state)) {
+        fetch(`${AUTH_API_BACKEND}/verify?email=${pendingEmail}&action=${state}`, {
+          headers: {
+            'Authorization': `bearer ${access_token}`,
+            'Content-Type': 'application/json',
+          },
+        }).then((response) => {
+          if (response.ok) {
+            return response.json();
+          } else {
+            throw new Error(`Fetch error: ${response.status}`);
+          }
+        }).then((data) => {
+          if (data && data.result === 'success') {
+            silentSignin();  // Update user data from oidc. Same as renew signin after 10 minutes.
```

Comment:
You don't need it here. Edo's api already do renew token. So you actually need to take it from user onject

Reply:
Now using API for request/verify. Using API access token and updating it properly.